list.py

```diff
@@ -0,0 +1,248 @@
+class Node:
+    def __init__(self, elem, next = None):
+        self.elem = elem
+        self.next = next
+
+
+class ListIterator:
+    def __init__(self, head):
+        self.current = head
+
+    
+    def __iter__(self):
+        return self
+
+
+    def __next__(self):
+        if self.current == None:
+            raise StopIteration
+
+        elem = self.current.elem
+        self.current = self.current.next
+        return elem
+
+
+class List:
+    def __init__(self):
+        self.head = None
+        self.tail = None
+        self.length = 0
+
+
+    def __len__(self):
+        return self.length
+
+
+    def __eq__(self, other: 'List'):
+        curr_node = self.head
+
+        count = 0
+
+        if self.length != other.length:
+            return False
+
+        while count < self.length:
+
+            if self[count] != other[count]:
+                return False
+
+            curr_node = curr_node.next
+            count += 1
+
+        return True
+
+    
+    def __str__(self) -> str:
+        string = '['
+
+        curr_node = self.head
+
+        while curr_node: # проверка на None
+            string += curr_node.elem + ', ' * (curr_node.next != None) 
+            curr_node = curr_node.next 
+        
+        string += ']'
+
+        return string
+
+
+    def __getitem__(self, index):
+        self.check_index_borders(index)
+        index = self.index_transform(index)
+
+        count = 0
+        for elem in self:
+            if count == index:
+                return elem
+            count += 1
+
+
+    def __setitem__(self, index, elem):
+        self.check_index_borders(index)
+        self.find_node(self.index_transform(index)).elem = elem
+        
+
+    def __iter__(self):
+        return ListIterator(self.head)
+
+
+    def index_transform(self, index: int) -> int:
+        '''
+        Transform negative indices to correct positive value
+        '''
+        return index + self.length if index < 0 else index
+
+
+    def check_index_borders(self, index, border = 1):
+        '''
+        check index in: 
+            [-len; len): border = 1 (default)
+            [-len; len]: border = 0
+        '''
+        if index > self.length - border or index < -self.length:
+            self.raise_exception_oob(index)
```

Comment:
I propose to squash both methods `index_transform` and `check_index_borders` into one because these methods are almost always used one after another.

So there could be a method that combines both raises an error in case if the index is out of bounds and returns the effective non-negative index value